hub/util/exceptions.py

```diff
@@ -316,3 +316,32 @@ def __init__(self, meta_key: str, expected: Any, actual: Any):
 class ReadOnlyModeError(Exception):
     def __init__(self):
         super().__init__("Modification when in read-only mode is not supported!")
+
+
+class InvalidTransformOutput(Exception):
+    def __init__(self):
+        super().__init__(
+            "The output of each step in a transformation should be either dictionary or a list/tuple of dictionaries."
+        )
+
+
+class InvalidInputDataException(Exception):
+    def __init__(self, message):
+        super().__init__(
+            f"The data_in to transform is invalid. It should support {message} operation."
+        )
+
+
+class UnsupportedSchedulerException(Exception):
+    def __init__(self, scheduler):
+        super().__init__(
+            f"Hub Transform currently doesn't support {scheduler} scheduler."
+        )
+
+
+class TensorMismatchException(Exception):
+    def __init__(self, tensors, output_keys):
+        super().__init__(
+            f"One or more of the outputs generated during transform contain different tensors than the ones present in the output 'ds_out' provided to transform.\n "
+            f"Tensors in ds_out: {tensors}\n Tensors in output sample: {output_keys}"
+        )
```

Comment:
can you create a base Error

```python
class TransformError(Exception):
    pass
```

and have all of these exceptions start with `Transform`, inherit from `TransformError`, and **end with Error**?

this is the current naming convention